lib/libc/math/lib_log.c

```diff
@@ -85,14 +105,23 @@ double log(double x)
 		}
 
 		epsilon = (fabs(y) > 1.0) ? fabs(y) * DBL_EPSILON : DBL_EPSILON;
+
+		if (++iter >= LOGF_MAX_ITER) {
+			relax_factor *= LOGF_RELAX_MULTIPLIER;
+			iter = 0;
+		}
+
+		if (relax_factor > 1) {
+			epsilon *= relax_factor;
+		}
 	}
 
 	if (y == 700.0) {
 		return INFINITY;
 	}
 
 	if (y == -700.0) {
-		return -INFINITY;
+		return INFINITY;
```

Comment:
In this case, how a differentation is made between very large and very small values?

Reply:
Reverting the change.